js/src/ui/TxList/txList.js

```diff
@@ -35,19 +35,16 @@ class TxList extends Component {
       PropTypes.array,
       PropTypes.object
     ]).isRequired,
-    netVersion: PropTypes.string.isRequired
+    blockNumber: PropTypes.object,
+    netVersion: PropTypes.string.isRequired,
+    newError: PropTypes.func
   };
 
-  store = new Store(this.context.api);
-
   componentWillMount () {
+    this.store = new Store(this.context.api, this.props.newError);
```

Comment:
Ok, so the issue was that this:
```js
store: new Store(this.context.api, this.props.newError)
```
doesn't work cause `this.props.newError` is not available yet because of the lifetime issues you mentioned. So your solution was to do:
```js
componentWillMount () {
  this.store = new Store(this.context.api, this.props.newError);
}
```
And that works correctly. My point is that it's still not working for all cases (it's not generic enough), cause `this.props.newError` can change if the parent component decides to change it. So proper implementation should look like this:
```js
componentWillMount () {
  this.store = new Store(this.context.api, this.props.newError);
}
componentWillReceiveProps (newProps) {
  if (this.props.newErrror !== newProps.newError) {
    this.store = new Store(this.context.api, newProps.newError);
    // + copy state of the previous store most probably.
  }
}
```
but that's way to complicated. So instead I proposed a simpler solution:
```js
store: new Store(this.context.api, (...args) => this.props.newError(...args))
```

So store has a reference to a lambda function that will dynamically dispatch to `this.props.newError` so even if `this.props.newError` is changed the Store doesn't need to be updated.

IMHO even if `this.props.newError` is not subject to change currently we shouldn't rely on that cause in future it may and such code may cause a wierd errors in the future that are easy to avoid now.

Hope this explains my motivations and solution ;)